test/core/tsi/crl_ssl_transport_security_test.cc

```diff
@@ -43,17 +43,22 @@ const int kSslTsiTestRevokedKeyCertPairsNum = 1;
 const int kSslTsiTestValidKeyCertPairsNum = 1;
 const char* kSslTsiTestCrlSupportedCredentialsDir =
     "test/core/tsi/test_creds/crl_data/";
+const char* kSslTsiTestFaultyCrlsDir = "bad_path/";
 
 class CrlSslTransportSecurityTest
     : public testing::TestWithParam<tsi_tls_version> {
  protected:
   // A tsi_test_fixture implementation.
   class SslTsiTestFixture {
    public:
+    // When use_faulty_crl_directory is set, the crl_directory of the
```

Comment:
Currently only adds the faulty directory to the client. I think this may be sufficient for testing the softfail but I can add another unit test for the server if need be.